Frontend/Gmfctn/src/app/modules/+auth/components/sign-in/sign-in.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { finalize, take } from 'rxjs/operators';
+import { AuthenticateService } from '../../services/authenticate.service';
+
+@Component({
+  selector: 'app-sign-in',
+  templateUrl: './sign-in.component.html',
+  styleUrls: ['./sign-in.component.scss'],
+})
+export class SignInComponent implements OnInit, OnDestroy {
+  @Output() isSigned = new EventEmitter<boolean>();
+  state = '';
+  loading = false;
+  subscribtion$: Subscription = new Subscription;
+  subscribtionForCorrectness$: Subscription = new Subscription;
+  userForm: FormGroup = new FormGroup({});
+  user: {userName: string, password: string} = {userName: '', password: ''};
```

Comment:
I'd suggest renaming it to `initialFormData`, also as you declaring the variable instantly I believe you can skip a type declaration as it'd be immediately casted

Reply:
Fixed